libs/libc/dirent/lib_nftw.c

```diff
@@ -0,0 +1,228 @@
+/****************************************************************************
+ * libs/libc/dirent/lib_nftw.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <ftw.h>
+#include <dirent.h>
+#include <sys/stat.h>
+#include <errno.h>
+#include <unistd.h>
+#include <string.h>
+#include <limits.h>
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int
+call_nftw(FAR char *path, nftw_cb_t fn, int flags, int base,
+          int level, FAR const struct stat *buf, int info)
+{
+  struct FTW ftw =
+  {
+    base, level
+  };
+
+  if (flags & FTW_CHDIR)
+    {
+      int r;
+
+      if (base > 1)
+        {
+          path[base - 1] = '\0';
+          r = chdir(path);
+          path[base - 1] = '/';
+        }
```

Comment:
no the caller must give the absolute path, otherwise lstat or stat at line 92 will fail first.

Reply:
should that be fixed?

Reply:
I don't know, because the spec don't see how to handle this? But if ftw/nftw need consider the current work directory, should we modify stat/lstat to support it too? Here is the musl implementation: 
https://github.com/bminor/musl/blob/master/src/misc/nftw.c
which don't do any special thing here.

Reply:
as the musl impl doesn't seem to do chdir at all, i guess relative path can just work there.

Reply:
musl can work, just because stat/lstat/chdir support the cwd concept, here is musl' lstat implementation:
https://github.com/bminor/musl/blob/master/src/stat/lstat.c#L6
you can see lstat use AT_FDCWD flag there.
So the correct fix is enhance the low level file system related functions in nuttx with the cwd support.

Reply:
ok. it makes sense.